pkg/localstore/mode_set.go

```diff
@@ -254,81 +236,128 @@ func (db *DB) setRemove(batch *leveldb.Batch, addr swarm.Address) (gcSizeChange
 	if err != nil {
 		return 0, err
 	}
-	err = db.gcIndex.DeleteInBatch(batch, item)
+	err = db.postageChunksIndex.DeleteInBatch(batch, item)
 	if err != nil {
 		return 0, err
 	}
+	// unless called by GC which iterates through the gcIndex
 	// a check is needed for decrementing gcSize
-	// as delete is not reporting if the key/value pair
-	// is deleted or not
-	if _, err := db.gcIndex.Get(item); err == nil {
-		gcSizeChange = -1
+	// as delete is not reporting if the key/value pair is deleted or not
+	if check {
+		_, err := db.gcIndex.Get(item)
+		if err != nil {
+			if !errors.Is(err, leveldb.ErrNotFound) {
+				return 0, err
+			}
+			return 0, db.pinIndex.DeleteInBatch(batch, item)
+		}
 	}
-
-	return gcSizeChange, nil
+	err = db.gcIndex.DeleteInBatch(batch, item)
+	if err != nil {
+		return 0, err
+	}
+	return -1, nil
 }
 
 // setPin increments pin counter for the chunk by updating
 // pin index and sets the chunk to be excluded from garbage collection.
 // Provided batch is updated.
-func (db *DB) setPin(batch *leveldb.Batch, addr swarm.Address) (err error) {
-	item := addressToItem(addr)
-
+func (db *DB) setPin(batch *leveldb.Batch, item shed.Item) (gcSizeChange int64, err error) {
 	// Get the existing pin counter of the chunk
-	existingPinCounter := uint64(0)
-	pinnedChunk, err := db.pinIndex.Get(item)
+	i, err := db.pinIndex.Get(item)
+	item.PinCounter = i.PinCounter
```

Comment:
what are we using this for ?

Reply:
you tell me https://github.com/ethersphere/bee/pull/1063/files#r553872136 :shrug: